src/Listeners/KernelEventsSubscriber.php

```diff
@@ -234,19 +238,23 @@ public function onKernelFinishRequest(FinishRequestEvent $event): void
      */
     public function onKernelException($event): void
     {
+        if (! $this->inspector->isRecording()) {
+            return;
+        }
         // Compatibility with Symfony < 5 and Symfony >=5
         // The additional `method_exists` check is to prevent errors in Symfony 4.3
         // where the ExceptionEvent exists and is used but doesn't implement
         // the `getThrowable` method, which was introduced in Symfony 4.4
         if ($event instanceof ExceptionEvent && \method_exists($event, 'getThrowable')) {
-            $this->startTransaction(get_class($event->getThrowable()))->setResult('error');
-            $this->notifyUnexpectedError($event->getThrowable());
+            $exception = $event->getThrowable();
         } elseif ($event instanceof GetResponseForExceptionEvent) {
-            $this->startTransaction(\get_class($event->getException()))->setResult('error');
-            $this->notifyUnexpectedError($event->getException());
+            $exception = $event->getException();
         } else {
             throw new \LogicException('Invalid exception event.');
         }
+
+        $this->startTransaction(get_class($exception))->setResult('error');
```

Comment:
This start transaction must be deleted for two reasons:

1. If there is an active transaction this new start will replace it;
2. If there isn't an active transaction the underlying library starts a new one by default: https://github.com/inspector-apm/inspector-php/blob/master/src/Inspector.php#L237